### PR TITLE
ENH: Implement write_train_log_for_tensorboard

### PR DESCRIPTION
**Pull request by Leengit**
Loss, accuracy, validation loss, and validation accuracy from `GenericModelHandler.CustomCallback.log` entries for `ON_TRAIN_END` events are written to a directory of files for use with `TensorBoard`.  This allows comparison of the training outcomes each time additional labels are used in active learning.

The interface to the new `write_train_log_for_tensorboard` method exactly matches the interface to [`torch.utils.tensorboard.SummaryWriter`](https://pytorch.org/docs/stable/tensorboard.html).  The written files can be used within `TensorBoard` with `tensorboard --logdir=runs`, where `runs` is the default name of the directory used by `SummaryWriter` and hence by `write_train_log_for_tensorboard`.

This pull request also implements `write_epoch_log_for_tensorboard`, which writes entries for `ON_TRAIN_EPOCH_END` events.  It is for situations where a single training is being examined for it changes from epoch to epoch.

### CODE REVIEW

File: al_bench/model.py [hunk 1]
```diff
@@ -136,31 +142,107 @@ def get_dataset_handler(self):
...
+                beginning = datetime.utcfromtimestamp(0)
+                for entry in self.log:
+                    if entry["model_step"] != model_step:
+                        continue
+                    logs = entry["logs"]
+                    if logs is None:
+                        continue
+                    utc_seconds = (entry["utcnow"] - beginning).total_seconds()
+                    x_value = entry[x_key]
+                    for key in y_dictionary.keys():
```
Comment (manthey) [inline al_bench/model.py:181]: Doesn't the standard dictionary iterator iterate over keys?  That is, I think `.keys()` is unnecessary here and in the line below.

Comment (Leengit) [reply al_bench/model.py:181]: Thank you.  I will change those.
